### PR TITLE
rename `Layer.plane` to `Layer.slicing_plane`

### PR DESCRIPTION
**Pull request by alisterburt**
# Description
This PR renames the `Layer.plane` attribute added in #3023 to `Layer.embedded_plane` for more clarity with respect to #3140

### CODE REVIEW

File: napari/layers/image/image.py [hunk 1]
```diff
@@ -498,8 +498,8 @@ def rendering(self, rendering):
         self.events.rendering()
 
     @property
-    def plane(self):
-        return self._plane
+    def embedded_plane(self):
+        return self._embedded_plane
 
     @plane.setter
```
Comment (Carreau) [inline napari/layers/image/image.py]: that setter will need to be removed, or set to ``@embeded_plane``

**Comment by sofroniewn** [comment]
I'm not sure I love the term `embedded_plane`, I still find myself guessing what it might be. Is that a standard term from other software / fields that I don't know about or something you made up @alisterburt?

I guess we need to distinguish between `clipping_plane(s)` which I think is clear from the name and the plane which is a projection of the data - of which there will only ever be one of at a time? Maybe `embedding_plane` is a good term, or maybe a `projection_plane` or maybe `slicing_plane`? That last one could be my favourite. Thoughts @alisterburt @jni?

I do agree that we probably need another word here to make things less confusing

**Comment by alisterburt** [comment]
It's definitely made up and not standard! I think I prefer `slicing_plane` too... will wait to see if @kevinyamauchi @brisvag @jni want to weigh in but if nothing comes up I'll update to `slicing_plane` 👍🏼 thanks!

**Comment by brisvag** [comment]
Agree on slicing plane being clearer!

**Comment by kevinyamauchi** [comment]
I also prefer `slicing_plane` to `embedded_plane`. My minor concern is that we use "slicing" to describe slicing the layer data in other parts of the code base. Perhaps `rendered_plane`, `rendering_plane`, or `render_plane`? However, I am fine with `slicing_plane` if that's the consensus!

**Comment by brisvag** [comment]
`projected_plane` or variation thereof can also be an option. (Oh, was already proposed :P)

**Comment by brisvag** [comment]
By the way, I think it's more convenient if I make this change as part of #3140 rather than merging this PR... otherwise I need to fix these conflicts for the nth time in a row :P

**Comment by alisterburt** [comment]
Thanks for your input guys, my order of preference is `slicing_plane` > `rendered_plane` > `projected_plane`. I think I like `slicing_plane` despite the name clash and think it's okay because 
- it's in a different place
- it's not just `slicing`, it includes the word plane

@brisvag totally agree that you should include this in #3140 !

**Comment by sofroniewn** [comment]
> By the way, I think it's more convenient if I make this change as part of #3140 rather than merging this PR... otherwise I need to fix these conflicts for the nth time in a row :P

While I know it's a pain to fix the merge conflicts constantly, it is much easier for us to review and merge smaller PRs and I think makes the overall development speed of the project faster. Sorry about this, but hopefully it's just one more update now!!

> I think I like slicing_plane despite the name clash and think it's okay because

Sounds like this is still the favourite, so let's go with it.

@alisterburt if you can quickly update this PR to `slicing_plane` and then we can merge and @brisvag only has to fix merge conflicts once more. Thanks!!

**Review by sofroniewn — APPROVED** [review]
LGTM! Let's merge this now, though see https://github.com/napari/napari/issues/3148#issuecomment-899654166 that we might want to make this still be experimental or private before `0.4.11` release. Let's do that later though if we decide necessary

**Comment by alisterburt** [comment]
updated, including the setter, thanks @Carreau ! this should be ready to go

**Comment by sofroniewn** [comment]
Tests still failing. @alisterburt based on convo in #3148 should we add `experimental` now before this merges? Might as well I think, so `layer.experimental_slicing_plane`

**Comment by alisterburt** [comment]
done but v quickly before heading out, haven't checked for issues!!

**Review by sofroniewn — APPROVED** [review]
Looks great! will merge now

**Comment by brisvag** [comment]
> While I know it's a pain to fix the merge conflicts constantly, it is much easier for us to review and merge smaller PRs and I think makes the overall development speed of the project faster. Sorry about this, but hopefully it's just one more update now!!

Hehe, fair enough, I was being lazy :P
I actually already did the same yesterday in my branch (but haven't pushed), and I think there might be a couple differences. Will pull this and see if any of us missed something!

**Pull request merged**